src/routes/[id]/+page.svelte

```diff
@@ -2,16 +2,18 @@
 	import type { FileData, UserData } from '$lib/types';
 	import { onValue, ref, remove, serverTimestamp, set, update } from 'firebase/database';
 	import IDE from './IDE.svelte';
-	import { authState, database } from '$lib/firebase/firebase.svelte';
+	import { authState, database, USER_DATA_KEY } from '$lib/firebase/firebase.svelte';
 	import { computePermission } from '$lib/utils';
 	import MessagePage from '$lib/components/MessagePage.svelte';
+	import { getContext } from 'svelte';
 
 	let props = $props();
 	let fileId = props.data.fileId;
 
 	let isLoading = $state(true);
 	let fileData: FileData | null = $state(null);
-	let userData: UserData | null = $state(null);
+	const userData: UserData = getContext(USER_DATA_KEY);
```

Comment:
You can consider exporting `getUserData()` as a function from `firebase.svelte` that calls getContext from there.

See https://svelte.dev/docs/svelte/context#Encapsulating-context-interactions